### DISCUSSION TITLE
Avoiding property name recursion

### DISCUSSION THREAD

**Discussion by mikea** [apple/pkl#602] [Q&A]
I often find myself writing the code like:

```
function myFactoryFn(name: String) = new mySomething {
  name = name
}
```

Which creates an infinite recursion. 

The way I deal with it is by renaming function parameter. 
Which is suboptimal from consistency and documentation perspective. 

Is there a better way to deal with the problem without naming every parameter aName or similar?

**Comment by bioball** [comment]
Renaming the parameter is the way to go here. The convention that our core team follows here is to prefix the parameter name with an underscore, e.g.

```pkl
function myFactoryFn(_name: String) = new mySomething {
  name = _name
}
```
